### PR TITLE
Initial structure, dep, MetricsBackend CRD and generated code

### PR DESCRIPTION
**Pull request by mattkelly**
This is just the initial project structure, including the first CRD (`MetricsBackend`) to test out code generation.

Mainly PRing now because @ashleyschuett has started to use this foundation as well.

### CODE REVIEW

**Comment by mattkelly** [comment]
I'll integrate Travis in a separate PR.

File: pkg/apis/metricsbackend.cerebral.containership.io/v1alpha1/doc.go [hunk 1]
```diff
@@ -0,0 +1,6 @@
+// +k8s:deepcopy-gen=package,register
+
+// Package v1alpha1 is the v1alpha1 version of the API.
+// +groupName=provision.containership.io
```
Comment (ashleyschuett) [inline pkg/apis/metricsbackend.cerebral.containership.io/v1alpha1/doc.go]:
```
// +groupName=metricsbackend.cerebral.containership.io
// +groupGoName=MetricsBackend
```
?

File: pkg/apis/metricsbackend.cerebral.containership.io/v1alpha1/types.go [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"k8s.io/apimachinery/pkg/selection"
+)
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
```
Comment (ashleyschuett) [inline pkg/apis/metricsbackend.cerebral.containership.io/v1alpha1/types.go]: is this going to be namespaces or should we add `// +genclient:nonNamespaced`

File: pkg/apis/metricsbackend.cerebral.containership.io/v1alpha1/doc.go (hunk 1 above)
Comment (mattkelly) [reply pkg/apis/metricsbackend.cerebral.containership.io/v1alpha1/doc.go]: Nice catch, updated.

File: pkg/apis/metricsbackend.cerebral.containership.io/v1alpha1/types.go (hunk 2 above)
Comment (mattkelly) [reply pkg/apis/metricsbackend.cerebral.containership.io/v1alpha1/types.go]: Nice catch, updated.

**Pull request merged**